### PR TITLE
Arm64:Implemeting ThisPtrRetBufPrecode:

### LINKED ISSUE #4284: ARM64: Implement ThisPtrRetBufPrecode::Init

### PR DESCRIPTION
**Pull request by ramarag**
Precode to shuffle this and retbuf for closed delegates over static methods with return buffer.

This resolves #4284 

@gkhanna79 @rahku PTAL
@dotnet/arm64-contrib FYI

### CODE REVIEW

**Comment by ramarag** [comment]
@dotnet-bot test Windows_NT arm64 Checked

**Comment by ramarag** [comment]
@kyulee1 how do I add a new test to the CI machine ?

**Comment by kyulee1** [comment]
There is not automatic way yet. See one note "ARM64 CI Overview-Arm64 machine setup".
Basically you should log on each machine and add new binary, and also update test zip file so that other folks can use it for the local testing.

File: src/vm/arm64/cgencpu.h [hunk 1]
```diff
@@ -614,39 +614,42 @@ struct FixupPrecode {
...
 
     BOOL SetTargetInterlocked(TADDR target, TADDR expected)
     {
-        _ASSERTE(!"ARM64:NYI");
-        return NULL;
+        CONTRACTL
+        {
+            THROWS;
+        GC_TRIGGERS;
+        }
```
Comment (rahku) [inline src/vm/arm64/cgencpu.h]: indentation

File: src/vm/arm64/stubs.cpp [hunk 2]
```diff
@@ -572,9 +572,27 @@ void FixupPrecode::Fixup(DataImage *image, MethodDesc * pMD)
...
 void ThisPtrRetBufPrecode::Init(MethodDesc* pMD, LoaderAllocator *pLoaderAllocator)
 {
-    _ASSERTE(!"ARM64:NYI");
+    WRAPPER_NO_CONTRACT;
+
+    int n = 0;
+    //Initially
+    //x0 -This ptr
+    //x1 -ReturnBuffer
+    m_rgCode[n++] = 0x91000010; // mov x16, x0
```
Comment (rahku) [inline src/vm/arm64/stubs.cpp]: Is this correct mov(register) (section C6.6.125 in arm64 manual) starts with encoding 0xa.......
which encoding is this? section(?)

Comment (ramarag) [inline src/vm/arm64/stubs.cpp]: this is the encoding produced by arm64 assembler

File: src/vm/arm64/cgencpu.h (hunk 1 above)
Comment (ramarag) [inline src/vm/arm64/cgencpu.h]: will correct

File: src/vm/arm64/stubs.cpp (hunk 2 above)
Comment (rahku) [inline src/vm/arm64/stubs.cpp]: ok

**Comment by rahku** [comment]
LGTM

**Comment by ramarag** [comment]
@dotnet-bot test Windows_NT arm64 Checked

**Pull request merged**